internal/pkg/composable/providers/kubernetessecrets/config.go

```diff
@@ -10,4 +10,12 @@ import "github.com/elastic/elastic-agent-autodiscover/kubernetes"
 type Config struct {
 	KubeConfig        string                       `config:"kube_config"`
 	KubeClientOptions kubernetes.KubeClientOptions `config:"kube_client_options"`
+
+	TTL string `config:"ttl"`
```

Comment:
Why are you making this a string? Make it a `time.Duration`. Elastic Agent uses https://github.com/elastic/go-ucfg which supports `time.Duration` as a native configuration type.